src/main/java/com/scrive/pdftools/tools/flattener/LambdaFlattener.java

```diff
@@ -0,0 +1,40 @@
+package com.scrive.pdftools.tools.flattener;
+
+import com.itextpdf.text.DocumentException;
+import com.itextpdf.text.exceptions.BadPasswordException;
+import com.itextpdf.text.pdf.PdfReader;
+import com.itextpdf.text.pdf.PdfStamper;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+public class LambdaFlattener {
+
+    public static ByteArrayOutputStream execute(FlattenerSpec flatSpec) throws IOException {
+
+        final ByteArrayInputStream file = new ByteArrayInputStream(flatSpec.mainFileInput.getContent());
```

Comment:
don't we want to protect the `file` in a try-with-resources?

Reply:
Yes. The code is  much nicer now :)

Reply:
Yes, fixed